### PR TITLE
fix(fast-foundation): turn off indeterminate on checked change

### LINKED ISSUE #6112: fix: checked and indeterminate co-exist in checkbox
# 🐛 Bug Report

setting `checked` to checkbox while in `indeterminate` state doesn't set off `indeterminate`.
natively, setting input of type checkbox to `checked` will set `indeterminate` to false.

## 💻 Repro or Code Sample

1. visit [checkbox page on fast explorer](https://explore.fast.design/components/fast-checkbox)
2. set `indeterminate` property to `true`
3. click to checkbox to be `checked`
4. it will set both `checked` & `indeterminate` to true

## 🤔 Expected Behavior

setting `checked` to checkbox while in `indeterminate` state should set `indeterminate` off.

## 💁 Possible Solution

programmatically observe and handle checkbox `checked` change and set `indeterminate` off if is on.

## 🔦 Context

briefly going over [w3c](https://www.w3schools.com/jsref/prop_checkbox_indeterminate.asp) -
> Checkboxes actually has three states: true, false and indeterminate which indicates that a checkbox is neither "on" or "off".

image

### PR DESCRIPTION
**Pull request by yinonov**
Closes #6112

# Pull Request

### General

- [x] I have included a change request file using `$ yarn change`
- [x] I have added tests for my changes.
- [ ] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [ ] I have read the [CONTRIBUTING](https://github.com/microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [x] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -3,7 +3,11 @@
     "files.insertFinalNewline": false
   },
   "[markdown]": {
-    "editor.quickSuggestions": false,
+    "editor.quickSuggestions": {
```
Comment (yinonov) [inline .vscode/settings.json:6]:
```suggestion
    "editor.quickSuggestions": false,
```

File: .vscode/settings.json [hunk 2]
```diff
@@ -4,6 +4,10 @@
   },
   "[markdown]": {
     "editor.quickSuggestions": false,
+      "comments": "off",
+      "strings": "off",
+      "other": "off"
+    },
```
Comment (yinonov) [inline .vscode/settings.json:10]:
```suggestion
```

File: change/@microsoft-fast-foundation-f16f025d-52ef-4c87-b19d-4b14310d2951.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "fast-foundation checkbox - turn off indeterminate on checked change",
```
Comment (yinonov) [inline change/@microsoft-fast-foundation-f16f025d-52ef-4c87-b19d-4b14310d2951.json:3]:
```suggestion
  "comment": "fast-foundation checkbox - turn off indeterminate on user `checked` change",
```

**Review by chrisdholt — APPROVED** [review]

**Comment by yinonov** [comment]
[discussed in discord](https://discord.com/channels/449251231706251264/482296787021856768/986734638682763284). seem to be aligned to expected behavior

**Review by EisenbergEffect — APPROVED** [review]

**Pull request merged**